src/component/axis/AxisBuilder.ts

```diff
@@ -781,46 +781,51 @@ function buildAxisLabel(
 
         const tickCoord = axis.dataToCoord(tickValue);
 
-        const font = labelModel.getFont();
-        const labelFormatter = makeLabelFormatter(axis);
-        const rect = textContain.getBoundingRect(
-            labelFormatter({ value: tickValue }), font, 'center', 'top'
-        );
-
-        // correct labels margin in the other side.
-        const labelMarginCorrected = interleaved && index % 2 ? labelMargin + rect.height : labelMargin;
-
-        const textEl = new graphic.Text({
-            x: tickCoord,
-            y: opt.labelOffset + opt.labelDirection * labelMarginCorrected,
-            rotation: labelLayout.rotation,
-            silent: silent,
-            z2: 10 + (labelItem.level || 0),
-            style: createTextStyle(itemLabelModel, {
-                text: formattedLabel,
-                align: itemLabelModel.getShallow('align', true)
-                    || labelLayout.textAlign,
-                verticalAlign: itemLabelModel.getShallow('verticalAlign', true)
-                    || itemLabelModel.getShallow('baseline', true)
-                    || labelLayout.textVerticalAlign,
-                fill: typeof textColor === 'function'
-                    ? textColor(
+        const textElStyle = createTextStyle(itemLabelModel, {
+            text: formattedLabel,
+            align: itemLabelModel.getShallow('align', true)
+                || labelLayout.textAlign,
+            verticalAlign: itemLabelModel.getShallow('verticalAlign', true)
+                || itemLabelModel.getShallow('baseline', true)
+                || labelLayout.textVerticalAlign,
+            fill: typeof textColor === 'function'
+                ? textColor(
                         // (1) In category axis with data zoom, tick is not the original
                         // index of axis.data. So tick should not be exposed to user
                         // in category axis.
                         // (2) Compatible with previous version, which always use formatted label as
                         // input. But in interval scale the formatted label is like '223,445', which
                         // maked user repalce ','. So we modify it to return original val but remain
                         // it as 'string' to avoid error in replacing.
-                        axis.type === 'category'
-                            ? rawLabel
-                            : axis.type === 'value'
+                    axis.type === 'category'
+                        ? rawLabel
+                        : axis.type === 'value'
                             ? tickValue + ''
                             : tickValue,
-                        index
-                    )
-                    : textColor as string
-            })
+                    index
+                )
+                : textColor as string
+        });
+
+        const textElForGettingRect = new graphic.Text({
```

Comment:
You can use the existing text element to get bounding rect. It's not necessary to create a new one.